syntax = "proto3";


package poc.user;


import "google/protobuf/empty.proto";


message User {
string id = 1;
string name = 2;
string email = 3;
}


message CreateUserRequest {
string name = 1;
string email = 2;
}


message GetUserRequest { string id = 1; }
message ListUsersRequest {}
message ListUsersResponse { repeated User users = 1; }


message UpdateUserRequest {
string id = 1;
string name = 2;
string email = 3;
}


message DeleteUserRequest { string id = 1; }


service UserService {
rpc CreateUser(CreateUserRequest) returns (User);
rpc GetUser(GetUserRequest) returns (User);
rpc ListUsers(ListUsersRequest) returns (ListUsersResponse);
rpc UpdateUser(UpdateUserRequest) returns (User);
rpc DeleteUser(DeleteUserRequest) returns (google.protobuf.Empty);
}